api/jobspb/jobs.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto3";
+
+package jobs;
+
+option go_package = "api/jobspb;jobspb";
+
+import "google/protobuf/duration.proto";
+import "google/protobuf/timestamp.proto";
+import "google/rpc/status.proto";
+
+// StartJob asks pmm-agent to start job.
+message StartJob {
+  // Echo is a simple echo job that can be delayed.
+  message Echo {
+    string message = 1;
+    google.protobuf.Duration delay = 2;
+  }
+  // Timeout for the job.
+  google.protobuf.Duration timeout = 1;
+  oneof job {
+    Echo echo = 10;
+  }
+}
+
+// StopJob asks pmm-agent to stop job.
+message StopJob {}
+
+// JobResult represents job result.
+message JobResult {
+  // Error contains job error message.
+  message Error {
+    string message = 1;
+  }
+  // Echo contains result for echo job.
+  message Echo {
+    string message = 1;
+  }
+  google.protobuf.Timestamp timestamp = 1;
+  oneof result {
+    Error error = 10;
+    Echo echo = 11;
+  }
+}
+
+// JobProgress represents job progress messages like percentage of completion, status updates, etc.
+message JobProgress {
+  // Echo contains echo job status update.
+  message Echo {
+    string status = 1;
+  }
+  google.protobuf.Timestamp timestamp = 1;
+  oneof result {
+    Echo echo = 10;
+  }
+}
+
+// AgentMessage is a container for the agent messages.
+message AgentMessage {
```

Comment:
Maybe it's better to include it here? https://github.com/percona/pmm/blob/release%2F2.16/api/agentpb/agent.proto#L336

Reply:
I don't get it, include what? Message payload on line 61, but it's all about jobs only.